dashboard/modules/actor/actor_head.py

```diff
@@ -181,6 +188,28 @@ def process_actor_data_from_pubsub(actor_id, actor_table_data):
             except Exception:
                 logger.exception("Error processing actor info from GCS.")
 
+    async def _cleanup_actors(self):
+        while True:
+            try:
+                if len(DataSource.actors) > MAX_ACTORS_TO_CACHE:
+                    logger.debug("Cleaning up dead actors from GCS")
+                    while len(DataSource.actors) > MAX_ACTORS_TO_CACHE:
+                        if not self.dead_actors_queue:
+                            logger.warning(
+                                f"More than {MAX_ACTORS_TO_CACHE} "
+                                "live actors are cached"
+                            )
+                            break
+                        actor_id = self.dead_actors_queue.popleft()
+                        actor = DataSource.actors.pop(actor_id)
+                        job_id = actor["jobId"]
+                        node_id = actor["address"]["rayletId"]
```

Comment:
I think this is not always available?

Reply:
`actor["address"].get("rayletId")`?

Reply:
in the old code, it is assumed to exist: https://github.com/ray-project/ray/blob/aadd82dcbd6bb0a8083550ef3edf39c98bf08ce0/dashboard/modules/actor/actor_head.py#L156

Reply:
We also has 

```
            if node_id != actor_consts.NIL_NODE_ID:
                node_actors = dict(DataSource.node_actors.get(node_id, {}))
                node_actors[actor_id] = actor_table_data
                DataSource.node_actors[node_id] = node_actors
```
So node_id == `actor_consts.NIL_NODE_ID` if it doesn't exist. 

Maybe we need the same check (otherwise the delete might have the key error)

Reply:
Also can you add a test for this case? You can do this in this way

```python3
# This actor is not schedulable, so raylet Id will never be assigned
actor = Actor.options(num_gpus=1).remote()
ray.kill(actor)

```

Reply:
I think we're safe because nodes are only added to the Datasource.actors if they do have actor_id, node_id, and job_id.

I'm only deleting things that have already been cached. Anyways, I'll add a lot more safety checks to be safe.